diff --git a/drivers/rknpu/rknpu_iommu.c b/drivers/rknpu/rknpu_iommu.c
index 3e05fd5e517e..3fa56624597b 100644
--- a/drivers/rknpu/rknpu_iommu.c
+++ b/drivers/rknpu/rknpu_iommu.c
@@ -478,8 +478,6 @@ int rknpu_iommu_switch_domain(struct rknpu_device *rknpu_dev, int domain_id)
 				      "failed to allocate iommu domain\n");
 			return -EIO;
 		}
-		// init domain iova_cookie
-		iommu_get_dma_cookie(dst_domain);
 
 		iommu_detach_device(src_domain, rknpu_dev->dev);
 		ret = iommu_attach_device(dst_domain, rknpu_dev->dev);
