diff --git a/drivers/usb/gadget/function/f_uvc.c b/drivers/usb/gadget/function/f_uvc.c
index 09076224b5ca..be172fdec7a9 100644
--- a/drivers/usb/gadget/function/f_uvc.c
+++ b/drivers/usb/gadget/function/f_uvc.c
@@ -1350,3 +1350,4 @@ static struct usb_function *uvc_alloc(struct usb_function_instance *fi)
 DECLARE_USB_FUNCTION_INIT(uvc, uvc_alloc_inst, uvc_alloc);
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Laurent Pinchart");
+MODULE_IMPORT_NS(DMA_BUF);
